src/sentry/workflow_engine/migration_helpers/alert_rule.py

```
@@ -511,6 +514,71 @@ def update_migrated_alert_rule(alert_rule: AlertRule, updated_fields: dict[str,
+    updated_detector_trigger_fields: dict[str, Any] = {}
+    updated_action_filter_fields: dict[str, Any] = {}
+    if "label" in updated_fields:
+        updated_detector_trigger_fields["condition_result"] = PRIORITY_MAP[alert_rule_trigger.label]
+        updated_action_filter_fields["comparison"] = PRIORITY_MAP[alert_rule_trigger.label]
+    if "alert_threshold" in updated_fields:
+        updated_detector_trigger_fields["comparison"] = updated_fields["alert_threshold"]
+
+    detector_trigger.update(**updated_detector_trigger_fields)
+    if updated_action_filter_fields:
+        action_filter.update(**updated_action_filter_fields)
+
+    return detector_trigger, action_filter
+
+
+def dual_update_migrated_alert_rule_trigger_action(
+    trigger_action: AlertRuleTriggerAction, updated_fields: dict[str, Any]
+) -> Action | None:
+    # update the trigger before calling this method so that we can reuse the get_action_type method: is this kosher?
+    alert_rule_trigger = trigger_action.alert_rule_trigger
+    # Check that we dual wrote this action
+    priority = PRIORITY_MAP.get(alert_rule_trigger.label, DetectorPriorityLevel.HIGH)
+    detector_trigger = get_detector_trigger(alert_rule_trigger, priority)
+    if detector_trigger is None:
+        logger.info(
+            "alert rule was not dual written, returning early",
+            extra={"alert_rule": alert_rule_trigger.alert_rule},
+        )
+        return None
+    aarta = ActionAlertRuleTriggerAction.objects.get(alert_rule_trigger_action=trigger_action)
+    action = aarta.action
+
+    updated_action_fields: dict[str, Any] = {}
+    action_type = get_action_type(trigger_action)
+    if not action_type:
+        logger.warning(
+            "Could not find a matching Action.Type for the trigger action",
+            extra={"alert_rule_trigger_action_id": trigger_action.id},
+        )
+        # Do we want to make this an exception instead?
```

yeah I think we should fail loudly here and when there is no detector trigger

---

I don't think we should fail loudly on detector trigger—we call the update/delete helpers even if an alert isn't flagged into dual write, so we should just quit early.